### ISSUE TITLE
Authorization: Bearer token

### ISSUE THREAD

**Issue by awatson1978** [Vermonster/fhir-kit-client#135]
## Expected Behavior

Hello!  Trying to use `fhir-kit-client` to create resources in an EHR sandbox environment, and having some problems with adding the Bearer token.  Here is the code we're trying to use:

```javascript
let fhirServerUrl = getFhirServer();
let access_token = getAccessToken();

const fhirKitClient = new Client({
  baseUrl: fhirServerUrl,
  customHeaders: {
    "Content-Type": "application/json",
    "Authorization": "Bearer " + access_token
  }
});

let response = await fhirKitClient.create({
  resourceType: "Patient",
  body: newPatient
});

console.log("response", response);
```

## Actual Behavior

Unfortunately, we're getting a 400 error and the catch() block is showing that the headers are not being set on our POST method.

```
{ response:
   { status: 400,
     data: { resourceType: 'OperationOutcome', issue: [Array] } },
  config:
   { method: 'post',
     url:
      'https://fhir.epic.com/interconnect-fhir-oauth/api/FHIR/R4/Patient',
     headers: undefined } }
```

Any suggestions on how we might be able to add that bearer token to the post command?  

## Specifications

  - Version:  1.6.1
  - Platform:  Node

**Comment by jpnarkinsky** [comment]
Hey there.  Did you have any luck with this?

**Comment by bkaney** [comment]
Missed this before @jpnarkinsky / @awatson1978 . It was built so set the token with a setter `bearerToken`. See: https://github.com/Vermonster/fhir-kit-client/blob/master/lib/client.js#L123

For example: 
```
const fhirKitClient = new Client({
  baseUrl: fhirServerUrl,
  customHeaders: {
    "Content-Type": "application/json"
  }
});

fhirKitClient.bearerToken = token;
```

FWIW, there are default (overridable) headers are set here: https://github.com/Vermonster/fhir-kit-client/blob/master/lib/http-client.js#L7

**Comment by bkaney** [comment]
Looking at this more, I think what you did should have worked.. as seen here https://github.com/Vermonster/fhir-kit-client/blob/master/lib/http-client.js#L246 custom headers override the auth header.

Maybe try this to see what is being build on your side (see the last few lines here):
```
let fhirServerUrl = getFhirServer();
let access_token = getAccessToken();

const fhirKitClient = new Client({
  baseUrl: fhirServerUrl,
  customHeaders: {
    "Content-Type": "application/json",
    "Authorization": "Bearer " + access_token
  }
});

let response = await fhirKitClient.create({
  resourceType: "Patient",
  body: newPatient
});

// console.log("response", response);

const { request, response } = Client.httpFor(response);
console.log(request);

```

**Comment by bkaney** [comment]
This was a logging problem. Fixed in 1.6.3. Give it a shot!
